src/app/register-email-form/register-email-form.component.ts

```diff
@@ -72,19 +84,38 @@ export class RegisterEmailFormComponent implements OnInit {
     private changeDetectorRef: ChangeDetectorRef,
     private notificationsService: NotificationsService,
   ) {
-
   }
 
   ngOnInit(): void {
+    const validators: ValidatorFn[] = [
+      Validators.required,
+      Validators.email,
+      // Regex pattern borrowed from HTML5 specs for a valid email address:
+      // https://html.spec.whatwg.org/multipage/input.html#valid-e-mail-address
+      Validators.pattern('^[a-zA-Z0-9.!#$%&\'*+\/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$')
+    ];
     this.form = this.formBuilder.group({
       email: new FormControl('', {
-        validators: [Validators.required,
-          // Regex pattern borrowed from HTML5 specs for a valid email address:
-          // https://html.spec.whatwg.org/multipage/input.html#valid-e-mail-address
-          Validators.pattern('^[a-zA-Z0-9.!#$%&\'*+\/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$')
-        ],
+        validators: validators,
       })
     });
+    this.validMailDomains = [];
+    this.configService.findByPropertyName('authentication-password.domain.valid')
```

Comment:
Could we add some new specs for this new validation code into `register-email-form.component.spec.ts`?  Codecov is noting that this new validation code isn't covered by tests & I'm not seeing updates to the existing email validation tests in that file.